cmd/agent/api/server.go

```diff
@@ -69,124 +68,64 @@ func StartServer(
 	invHost inventoryhost.Component,
 	secretResolver secrets.Component,
 ) error {
-	err := initializeTLS()
+	apiAddr, err := getIPCAddressPort()
 	if err != nil {
-		return fmt.Errorf("unable to initialize TLS: %v", err)
+		return fmt.Errorf("unable to get IPC address and port: %v", err)
 	}
 
-	// get the transport we're going to use under HTTP
-	listener, err = getListener()
-	if err != nil {
-		// we use the listener to handle commands for the Agent, there's
-		// no way we can recover from this error
-		return fmt.Errorf("Unable to create the api server: %v", err)
-	}
+	extraHosts := []string{apiAddr}
 
-	err = util.CreateAndSetAuthToken()
-	if err != nil {
-		return err
-	}
+	ipcHost := config.Datadog.GetString("agent_ipc_host")
+	apiConfigPort := config.Datadog.GetInt("agent_ipc_port")
+	apiConfigEnabled := apiConfigPort > 0
+	apiConfigHostPort := net.JoinHostPort(ipcHost, strconv.Itoa(apiConfigPort))
 
-	// gRPC server
-	authInterceptor := grpcutil.AuthInterceptor(parseToken)
-	opts := []grpc.ServerOption{
-		grpc.Creds(credentials.NewClientTLSFromCert(tlsCertPool, tlsAddr)),
-		grpc.StreamInterceptor(grpc_auth.StreamServerInterceptor(authInterceptor)),
-		grpc.UnaryInterceptor(grpc_auth.UnaryServerInterceptor(authInterceptor)),
+	if apiConfigEnabled {
+		extraHosts = append(extraHosts, apiConfigHostPort)
```

Comment:
AFAIU, TLS x509 certs do not care about ports so we could/should probably just use the ipcHost

Reply:
Changed !